### PR TITLE
feat(M7): no-show tracking — cron job and service layer (KIM-329, KIM-333, KIM-334)

### PR DESCRIPTION
**Pull request by Oiranca**
## Summary

- Adds `mark_no_show_reservations()` PostgreSQL function (migration `20260412000000`) that marks `pending` reservations as `no_show` when their `end_time` has passed
- Adds `markNoShowReservations()` service function in `lib/server/reservations-service.ts` calling the RPC via the admin client
- Adds `POST /api/cron/mark-no-show` route handler following the same pattern as `/api/cron/cancel-pending` (CRON_SECRET auth, structured audit logging)
- Updates `lib/supabase/types.ts` to register the new RPC under `public.Functions`

## Test plan

- [ ] `pnpm typecheck` passes
- [ ] `pnpm build` passes (route `/api/cron/mark-no-show` appears in build output)
- [ ] `pnpm test` — 331 tests pass (no regressions)
- [ ] `POST /api/cron/mark-no-show` with correct `Authorization: Bearer <CRON_SECRET>` returns `{ marked: N }`
- [ ] `POST /api/cron/mark-no-show` without / with wrong token returns `401`
- [ ] qa-engineer to add unit tests for route handler and `markNoShowReservations()` (test files not created by backend-developer per test ownership rules)

Closes KIM-329, KIM-333, KIM-334

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: supabase/migrations/20260412000000_fn_mark_no_show_reservations.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+SET search_path = public, pg_catalog
+AS $$
+DECLARE
+  updated_count integer;
+BEGIN
+  UPDATE public.reservations
+  SET status = 'no_show'
+  WHERE status = 'pending'
+    AND activated_at IS NULL
+    AND (date::date + end_time::time) < NOW();
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply supabase/migrations/20260412000000_fn_mark_no_show_reservations.sql:25]: Fixed. Added partial index `reservations_pending_no_show_idx ON public.reservations (date, end_time) WHERE status = 'pending' AND activated_at IS NULL` to the migration, matching the pattern of `reservations_pending_date_idx` used for the cancel-expired cron.

File: lib/supabase/types.ts [hunk 2]
```diff
@@ -186,6 +186,7 @@ export type Database = {
     Functions: {
       cancel_expired_pending_reservations: { Args: { grace_minutes?: number }; Returns: number }
       is_admin: { Args: never; Returns: boolean }
+      mark_no_show_reservations: { Args: Record<string, never>; Returns: number }
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/supabase/types.ts]: Fixed. Changed `Args: Record<string, never>` to `Args: never` for `mark_no_show_reservations`, consistent with `is_admin` and all other no-arg RPCs in the types file.

File: lib/server/reservations-service.ts [hunk 3]
```diff
@@ -455,6 +455,13 @@ export async function cancelExpiredPendingReservations(): Promise<number> {
   return (data as number | null) ?? 0
 }
 
+export async function markNoShowReservations(): Promise<number> {
+  const admin = createSupabaseServerAdminClient()
+  const { data, error } = await admin.rpc('mark_no_show_reservations')
+  if (error) serviceError('Internal server error', 500)
+  return (data as number | null) ?? 0
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply lib/server/reservations-service.ts:462]: Fixed. Added tests for `markNoShowReservations` in `__tests__/server/reservations-service.test.ts` covering: happy path (RPC called, returns count), error path (RPC error → throws `ServiceError`), and null-data branch (`(data as number | null) ?? 0` → returns 0).

File: app/api/cron/mark-no-show/route.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        event: 'cron.mark_no_show_reservations.error',
+        timestamp: new Date().toISOString(),
+        error: err instanceof Error ? err.name : 'UnknownError',
+        ...(process.env.NODE_ENV !== 'production' && {
+          detail: err instanceof Error ? err.message : String(err),
+        }),
+      }),
+    )
+    return NextResponse.json({ error: 'Internal server error' }, { status: 500 })
+  }
```
[the remark this answers is not in the document]
Comment (Oiranca) [reply app/api/cron/mark-no-show/route.ts:31]: Fixed. Added `__tests__/app/api/cron/mark-no-show.test.ts` covering: missing/invalid Authorization header → 401, `CRON_SECRET` unset → 401, success → 200 with `{ marked: <number> }`, service error → 500, and zero-count edge case.
